packages/amplify-category-hosting/index.js

```diff
@@ -58,9 +59,17 @@ async function configure(context) {
 
 function publish(context, service, args) {
   const { enabledServices } = categoryManager.getCategoryStatus(context);
-
   if (enabledServices.length > 0) {
     if (enabledServices.includes(service)) {
+      if (service === 'S3AndCloudFront' && context.exeInfo?.amplifyMeta.hosting.S3AndCloudFront.output) {
+        if (context.exeInfo.amplifyMeta.hosting.S3AndCloudFront.output['CloudFrontSecureURL'] === undefined) {
```

Comment:
This condition might read a little better if you assign the CFN output to a variable then do `if (typeof cfnOutput === 'object' && cfnOutput.CloudFrontSecureURL === undefined`